spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -59,12 +67,49 @@ public DefaultSubscriberFactory(GcpProjectIdProvider projectIdProvider,
 	}
 
 	@Override
-	public Subscriber getSubscriber(String subscriptionName, MessageReceiver receiver) {
+	public Subscriber createSubscriber(String subscriptionName, MessageReceiver receiver) {
 		return Subscriber.newBuilder(
 				SubscriptionName.of(this.projectId, subscriptionName), receiver)
 				.setChannelProvider(this.channelProvider)
 				.setExecutorProvider(this.executorProvider)
 				.setCredentialsProvider(this.credentialsProvider)
 				.build();
 	}
+
+	@Override
+	public PullRequest createPullRequest(String subscriptionName, Integer maxMessages,
+			Boolean returnImmediately) {
+		PullRequest.Builder pullRequestBuilder =
+				PullRequest.newBuilder().setSubscriptionWithSubscriptionName(
+						SubscriptionName.of(this.projectId, subscriptionName));
```

Comment:
I guess `subscriptionName` cannot be empty. Let's ensure that with an appropriate `Assert.notEmpy()` in the begging of the method! Unless you can confirm that this `SubscriptionName.of()` does something on the matter to warn and end-user about  the mistake.